contracts/utils/Strings.sol

```diff
@@ -113,4 +128,261 @@ library Strings {
+     *
+     * Requirements:
+     * - The result must fit into an `uint256` type.
+     */
+    function tryParseHex(string memory input) internal pure returns (bool success, uint256 value) {
+        return tryParseHex(input, 0, bytes(input).length);
+    }
+
+    /**
+     * @dev Variant of {parseHex-string-uint256-uint256} that returns false if the parsing fails because of an
+     * invalid character.
+     *
+     * Requirements:
+     * - The result must fit into an `uint256` type.
+     */
+    function tryParseHex(
+        string memory input,
+        uint256 begin,
+        uint256 end
+    ) internal pure returns (bool success, uint256 value) {
+        bytes memory buffer = bytes(input);
+
+        // skip 0x prefix if present
+        bool hasPrefix = bytes2(buffer.unsafeReadBytesOffset(begin)) == bytes2("0x");
+        uint256 offset = hasPrefix.toUint() * 2;
+
+        uint256 result = 0;
+        for (uint256 i = begin + offset; i < end; ++i) {
+            uint8 chr = _tryParseChr(buffer[i]);
+            if (chr > 15) return (false, 0);
+            result *= 16;
+            result += chr;
+        }
+        return (true, result);
+    }
+
+    /**
+     * @dev Parse a hexadecimal string (with or without "0x" prefix), and returns the value as an `address`.
+     *
+     * Requirements:
+     * - The string must be formatted as `(0x)?[0-9a-fA-F]{40}`
+     */
+    function parseAddress(string memory input) internal pure returns (address) {
+        return parseAddress(input, 0, bytes(input).length);
+    }
+
+    /**
+     * @dev Variant of {parseAddress} that parses a substring of `input` located between position `begin` (included) and
+     * `end` (excluded).
+     *
+     * Requirements:
+     * - The substring must be formatted as `(0x)?[0-9a-fA-F]{40}`
+     */
+    function parseAddress(string memory input, uint256 begin, uint256 end) internal pure returns (address) {
+        (bool success, address value) = tryParseAddress(input, begin, end);
+        if (!success) revert StringsInvalidAddressFormat();
+        return value;
+    }
+
+    /**
+     * @dev Variant of {parseAddress-string} that returns false if the parsing fails because the input is not a properly
+     * formatted address. See {parseAddress} requirements.
+     */
+    function tryParseAddress(string memory input) internal pure returns (bool success, address value) {
+        return tryParseAddress(input, 0, bytes(input).length);
+    }
+
+    /**
+     * @dev Variant of {parseAddress-string-uint256-uint256} that returns false if the parsing fails because input is not a properly
+     * formatted address. See {parseAddress} requirements.
+     */
+    function tryParseAddress(
+        string memory input,
+        uint256 begin,
+        uint256 end
+    ) internal pure returns (bool success, address value) {
+        // check that input is the correct length
+        bool hasPrefix = bytes2(bytes(input).unsafeReadBytesOffset(begin)) == 0x3078;
```

Comment:
Using prefix `0x` as bytes2 multiple times. Could we declare it as a constant for readability?

Reply:
its just twice. I'm not sure having a constant would really help readability here.

Reply:
Switching to `bytes2("0x")` is preferred given https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5166#discussion_r1797359676. I see that was done.